xapian-core/api/queryinternal.cc

```diff
@@ -865,98 +867,156 @@ Query::Internal::unserialise(const char ** p, const char * end,
 	    return new Xapian::Internal::QueryValueRange(slot, begin, end_);
 	}
 	case 0: {
-	    // Other operators
-	    //
-	    //   000ttttt where:
-	    //     ttttt -> encodes which OP_XXX
-	    switch (ch & 0x1f) {
-		case 0x00: // OP_INVALID
-		    return new Xapian::Internal::QueryInvalid();
-		case 0x0a: { // Edit distance
-		    Xapian::termcount max_expansion;
-		    if (!unpack_uint(p, end, &max_expansion) || end - *p < 2) {
-			throw SerialisationError("not enough data");
-		    }
-		    int flags = static_cast<unsigned char>(*(*p)++);
-		    op combiner = static_cast<op>(*(*p)++);
-		    unsigned edit_distance;
-		    size_t fixed_prefix_len;
-		    string pattern;
-		    if (!unpack_uint(p, end, &edit_distance) ||
-			!unpack_uint(p, end, &fixed_prefix_len) ||
-			!unpack_string(p, end, pattern)) {
-			throw SerialisationError("not enough data");
+	    switch (ch >> 4) {
+	    case 1: {
+		// 0001oxxx where:
+		// o is the operation: LT or GT
+		// xxx is the slot number
+		Xapian::valueno slot = ch & 7;
+		if (slot == 7) {
+		    if (!unpack_uint(p, end, &slot)) {
+			unpack_throw_serialisation_error(*p);
+		}
+		slot += 7;
+		}
+		string begin;
+		if (!unpack_string(p, end, begin)) {
+		    unpack_throw_serialisation_error(*p);
+		}
+		switch (ch >> 3)
+		{
```

Comment:
We 'cuddle' opening braces for switch statements (as indeed you've done up at line 870).